src/mongo/db/storage/wiredtiger/encryption_keydb.cpp

```diff
@@ -421,17 +438,31 @@ void EncryptionKeyDB::clone(EncryptionKeyDB *old) {
 
 void EncryptionKeyDB::store_masterkey() {
     auto encodedKey = base64::encode(StringData{(const char*)_masterkey, _key_len});
-    if (!encryptionGlobalParams.kmipServerName.empty()) {
-        kmipWriteKey(encodedKey);
+    if (!encryptionGlobalParams.kmipServerName.empty() && _kmipMasterKeyId.empty()) {
+        _kmipMasterKeyId = kmipWriteKey(encodedKey);
+        LOGV2(29109,
```

Comment:
Don't forget to update [LOGV2 codes](https://confluence.percona.com/display/PSMDB/Document+logging+subsystem+identifiers+used+in+Percona+code) after merging the PR.